### PR TITLE
Fix normals

### PR DESCRIPTION
**Pull request by davepagurek**
Fixes https://github.com/calder-gl/calder/issues/72 by creating the matrix described in the issue. gl-matrix has a method `normalFromMat4` that encapsulates all the math, hence the lack of tests. I just added this matrix to the results returned from `traverse`.

### CODE REVIEW

File: src/armature/Node.ts [hunk 1]
```diff
@@ -355,18 +365,23 @@ export class Node {
...
             ),
 
-            vec3.fromValues(
-                Math.sqrt(
-                    Math.pow(position[0], 2) + Math.pow(position[1], 2) + Math.pow(position[2], 2)
-                ),
-                1,
-                1
-            )
+            vec3.fromValues(Math.max(1e-6, vec3.length(position)), 1, 1)
```
Comment (armcburney) [inline src/armature/Node.ts]: Why did you impose a max on this?

**Review by armcburney — APPROVED** [review]
LGTM! Nice work!

File: src/armature/Node.ts (hunk 1 above)
Comment (davepagurek) [reply src/armature/Node.ts]: If you have a child node that is also at the root of the parent, then the bone connecting them has length 0 and can't be inverted. This is to make sure the bone doesn't entirely disappear into two dimensions.

Comment (armcburney) [reply src/armature/Node.ts]: Ahh, sounds good to me, that makes sense. Could you maybe copy paste the above comment in the code so future viewers understand the decision too?

**Pull request merged**